src/Datadog.Trace/Agent/StreamFactories/NamedPipeClientStreamFactory.cs

```diff
@@ -0,0 +1,38 @@
+using System.IO;
+using System.IO.Pipes;
+
+namespace Datadog.Trace.Agent.StreamFactories
+{
+    internal class NamedPipeClientStreamFactory : IStreamFactory
+    {
+        private readonly string _pipeName;
+        private readonly string _serverName;
+        private readonly PipeOptions _pipeOptions;
+        private readonly int _timeoutMs;
+
+        public NamedPipeClientStreamFactory(string pipeName, int timeoutMs)
+            : this(pipeName, ".", PipeOptions.Asynchronous, timeoutMs)
+        {
+        }
+
+        public NamedPipeClientStreamFactory(string pipeName, string serverName, PipeOptions pipeOptions, int timeoutMs)
+        {
+            _pipeName = pipeName;
+            _serverName = serverName;
+            _pipeOptions = pipeOptions;
+            _timeoutMs = timeoutMs;
+        }
+
+        public string Info()
+        {
+            return $@"\\\\{_serverName}\\pipe\\{_pipeName}";
+        }
+
+        public Stream GetBidirectionalStream()
+        {
+            var pipeStream = new NamedPipeClientStream(_serverName, _pipeName, PipeDirection.InOut, _pipeOptions);
```

Comment:
This is getting called by `HttpStreamRequestFactory.Create`, which is called for each request to the trace agent. Is creating a new `NamedPipeClientStream` each time wasteful? Or is it reusing resources under the hood?

Reply:
If we find that the overhead is not substantial (as it runs once per second on average) would you be amenable to adding stream reuse in a later PR after we start to gather some numbers?

Reply:
Sounds good to me